src/Messages.sol

```diff
@@ -135,4 +135,29 @@ library ConnectorMessages {
         price = uint256(_msg.index(25, 32));
     }
 
+     /**
+     * Transfer
+     * 
+     * 0: call type (uint8 = 1 byte)
+     * 1-8: poolId (uint64 = 8 bytes)
+     * 9-25: trancheId (16 bytes)
+     * 26-46: user (Ethereum address, 20 bytes)
+     * 47-78: amount (uint256 = 32 bytes)
+     * 
+     */
+    function formatTransfer(uint64 poolId, bytes16 trancheId, address user, uint256 amount) internal pure returns (bytes memory) {
+        return abi.encodePacked(uint8(Call.Transfer), poolId, trancheId, user, amount);
+    }
+
+    function isTransfer(bytes29 _msg) internal pure returns (bool) {
+        return messageType(_msg) == Call.Transfer;
+    }
+
+    function parseTransfer(bytes29 _msg) internal pure returns (uint64 poolId, bytes16 trancheId, address user, uint256 amount) {
```

Comment:
I think the transfer message should probably contain the domain. For now, transfers will only go from a connector to cent chain and vice verse. But in the future, transfers from one domain to another (non cent chain one) will be possible, but they will always still go through cent chain. So cent chain will need to know the destination domain in order to forward it correctly.

Reply:
It seems that this is still missing the `domain` here as @offerijns mentioned

Reply:
Then I think it should be `formatTransfer` that adds the domain right? Since it's on the receiving end, `parseTransfer` shouldn't care about the destination domain, since it's already arrived and been validated by cent-chain.